### PR TITLE
Fix metavar anno

### PR DESCRIPTION
**Pull request by colleend**
fix the js file that couldn't be parsed

### CODE REVIEW

File: semgrep-core/tests/js/metavar_anno.js [hunk 1]
```diff
@@ -1,21 +1,29 @@
 // ERROR:
 @Anno1
 class Foo {
-    var foo = 1;
+    foo() {
+        var foo = 1;
+i    }
```
Comment (Sjord) [inline semgrep-core/tests/js/metavar_anno.js]: Is this `i` supposed to be here?

**Comment by mschwager** [comment]
I suspect the reason CI continued to pass despite this code parsing error was because the `exit` call got commented out: https://github.com/returntocorp/semgrep/commit/473fde80a85cad61a80dc4dfba35a55885fb37f4#diff-dd4b48a4893b775477afafe3626f023fc501d3977c707cacd892b903791c6d5cL155. Should we fix that too?

**Comment by aryx** [comment]
The CI is passing because we are now able to skip some parse errors and recover from it thx to tree-sitter, so
if you do 'make test' in semgrep-core you'll see that we correctly find the match even though there's a parsing
error later in the file :) 
```
Done: 0/0 (jobs: 0)pad@yrax:~/semgrep$ make test
./scripts/make-version > ./cli/version.ml
dune build
Done: 0/0 (jobs: 0)dune build ./_build/default/tests/test.bc
Done: 0/0 (jobs: 0)dune runtest -f
        Test alias pfff/tests/runtest
..............................(ONCE) TODO:ruby/parsing/heredoc.rb: adjust info out-of-bounds
...............
Ran: 45 tests in: 1.55 seconds.
OK
        test alias tests/runtest
.......................................................WARNING: fail to fully parse ../../../tests/js/metavar_anno.js
......................................................WARNING: fail to fully parse ../../../tests/js/metavar_anno.js
.....................................................WARNING: fail to fully parse ../../../tests/c/misc_ifdef_strings.c
.(Toplevel
   (MacroTop (("SOME_FUNC", ()), ((), [((Arg (C (Int ("1", ())))), [])], ()),
      (Some ()))))
PB: Ast_c_build, on ../../../tests/c/misc_toplevel_macrocall.c (exn = Common.Todo)
...............................................................................................................................................
Ran: 562 tests in: 4.07 seconds.
OK
Done: 3781/3783 (jobs: 1)pad@yrax:~/semgrep$ make install

```

**Comment by colleend** [comment]
makes sense 👍 i'll get rid of the i, good catch 👍

**Review by mschwager — APPROVED** [review]

**Pull request merged**